From 2511df692a8e230a2f6d00b76be3d9a9f791a459 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20Cz=C3=A9m=C3=A1n?= <trabarni@gmail.com>
Date: Wed, 16 Oct 2024 22:53:32 +0200
Subject: [PATCH] doc/dts: msm8974: Remove hammerhead from qcdtbs and document
 the quirks

LG Nexus 5 (hammerhead) should not be added for QCDTBS list it breaks
other devices. It works with ADTBS and QCDTBS list should be empty.
---
 Documentation/devices.md                            | 2 +-
 lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts | 5 +++++
 lk2nd/device/dts/msm8974/rules.mk                   | 1 -
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/Documentation/devices.md b/Documentation/devices.md
index 86162199b..5b71f6bf2 100644
--- a/Documentation/devices.md
+++ b/Documentation/devices.md
@@ -121,7 +121,7 @@
 - Fairphone 2 - FP2
 - HTC One M8 - m8
 - LG G3 - D855
-- LG Google Nexus 5 - hammerhead D820, D821
+- LG Google Nexus 5 - hammerhead D820, D821 (quirky - see comment in `lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts`)
 - Motorola Moto X 2014 - victara
 - OnePlus One - bacon <!--(use `lk2nd-msm8974-appended-dtb.img`)-->
 - Samsung Galaxy Note 3 - SM-N9005
diff --git a/lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts b/lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts
index 10104253e..ea1790961 100644
--- a/lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts
+++ b/lk2nd/device/dts/msm8974/msm8974-lge-hammerhead.dts
@@ -3,6 +3,11 @@
 #include <skeleton64.dtsi>
 #include <lk2nd.dtsi>
 
+/*
+ * To build for hammerhead, add LK2ND_QCDTBS="" to your make cmdline
+ * hammerhead does not work with qcdtbs enabled.
+ */
+
 / {
 	qcom,msm-id = <QCOM_ID_MSM8974 0x96 0x20002 0xb>;
 };
diff --git a/lk2nd/device/dts/msm8974/rules.mk b/lk2nd/device/dts/msm8974/rules.mk
index 1a2ffc24b..1358585b6 100644
--- a/lk2nd/device/dts/msm8974/rules.mk
+++ b/lk2nd/device/dts/msm8974/rules.mk
@@ -7,7 +7,6 @@ QCDTBS += \
 	$(LOCAL_DIR)/msm8974pro-ac-pm8941-mtp.dtb \
 	$(LOCAL_DIR)/msm8974-htc-m8.dtb \
 	$(LOCAL_DIR)/msm8974-lge-d855.dtb \
-	$(LOCAL_DIR)/msm8974-lge-hammerhead.dtb \
 
 ADTBS += \
 	$(LOCAL_DIR)/msm8974-lge-hammerhead.dtb \
-- 
2.47.0

